src/System.Text.RegularExpressions/tests/Regex.Cache.Tests.cs

```diff
@@ -76,14 +84,46 @@ public void Ctor_Cache_Promote_entries_does_not_throw()
                 Assert.True(Regex.IsMatch("1", "1"));
                 Assert.True(Regex.IsMatch("2", "2")); 
                 Assert.True(Regex.IsMatch("3", "3"));
+                Assert.True(GetCachedItemsNum() == 3);
                 Assert.True(Regex.IsMatch("1", "1")); // should be put first
+                Assert.True(GetCachedItemsNum() == 3);
                 Regex.CacheSize = 1;  // only 1 stays
+                Assert.True(GetCachedItemsNum() == 1);
             }
             finally
             {
                 Regex.CacheSize = originalCacheSize;
             }
         }
 
+        [Fact]
+        public void Ctor_Cache_Uses_culture_and_options()
+        {
+            int originalCacheSize = Regex.CacheSize;
+
+            try
+            { 
+                Regex.CacheSize = 0;
+                Regex.CacheSize = 3;
+                Assert.True(Regex.IsMatch("1", "1", RegexOptions.IgnoreCase));
+                Assert.True(Regex.IsMatch("1", "1", RegexOptions.Multiline));
+                Assert.True(GetCachedItemsNum() == 2);
+                CultureInfo.CurrentCulture = CultureInfo.GetCultureInfo("de-DE");
```

Comment:
Ah, I forgot there was no parameter for Culture. You have to set it on the entire thread. When running tests tests in UWP, currenlty this sets it for the whole process, and can mess up tests running concurrently.  To solve this you use RemoteInvoke. See the regex test GroupsEnUS for an example. Note you have to add RemoteExecutorTestBase as a base class for this test class.

Reply:
Then you don't need the 'finally' because the test sub-process just quits.

Reply:
Right, missed that :( Forgot about how tests run. To keep in mind for future.